### PR TITLE
Pro 2676 notify new email backend

### PR DESCRIPTION
**Pull request by tomxelliott**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by tomxelliott** [comment]
To be merged along with change to Business Service

File: app/components/services.js [hunk 1]
```diff
@@ -136,17 +136,10 @@ const updateInviteData = (inviteId, data) => {
     return utils.fetchJson(findInviteLinkUrl, fetchOptions);
 };
 
-const sendInvite = (data, sessionID) => {
+const sendInvite = (data, sessionID, exec, inviteId) => {
```
[the remark this answers is not in the document]
Comment (tomxelliott) [reply app/components/services.js]: Will be refactored at a later date as discussed. Would require a larger rewrite of a function that is out of the scope of the re-send ticket.

File: test/unit/testExecutorsWrapper.js [hunk 2]
```diff
@@ -462,4 +462,61 @@ describe('Executors.js', () => {
...
+                {firstName: 'james', lastName: 'miller', isApplying: true, isApplicant: true},
+                {fullname: 'ed brown', hasOtherName: true},
+                {fullname: 'jake smith'},
+                {fullname: 'bob smith'}
+            ]);
+            done();
+        });
+
+        describe('should return an empty list', () => {
+            it('when no executors emails have changed', (done) => {
```
Comment (shepda) [inline test/unit/testExecutorsWrapper.js]: This test shouldn't be in this `describe` block as it doesn't return an empty list.

File: app/steps/ui/executors/updateinvitesent/index.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+const Step = require('app/core/steps/Step');
+
+class ExecutorsUpdateInviteSent extends Step {
+
+    static getUrl() {
```
Comment (shepda) [inline app/steps/ui/executors/updateinvitesent/index.js]: Can you add a unit test for this method

File: app/steps/ui/executors/updateinvite/index.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    getContextData(req) {
+        const ctx = super.getContextData(req);
+        const formdata = req.session.form;
+        const executorsWrapper = new ExecutorsWrapper(formdata.executors);
+        ctx.executorsEmailChangedList = executorsWrapper.executorsEmailChangedList();
+        ctx.notifyExecutorsSuffix = ctx.executorsEmailChangedList.length > 1 ? '-multiple' : '';
+        return ctx;
+    }
+
+    action(ctx, formdata) {
```
Comment (shepda) [inline app/steps/ui/executors/updateinvite/index.js]: Can you add a unit test for this method

File: app/steps/ui/executors/contactdetails/index.js [hunk 5]
```diff
@@ -32,6 +33,7 @@ class ExecutorContactDetails extends ValidationStep {
     }
 
     * handlePost(ctx, errors) {
```
Comment (shepda) [inline app/steps/ui/executors/contactdetails/index.js]: Can you add a unit test for this method.

It doesn't appear that this class has any unit tests so you'll need to create the unit test file...don't worry about unit testing the methods that you've not changed.

File: app/steps/ui/declaration/index.js [hunk 6]
```diff
@@ -205,6 +208,8 @@ module.exports = class Declaration extends ValidationStep {
 
         delete ctx.executorsWrapper;
         delete ctx.hasDataChanged;
+        delete ctx.hasEmailChanged;
+        delete ctx.executorsEmailChanged;
         delete ctx.hasDataChangedAfterEmailSent;
         delete ctx.invitesSent;
         return [ctx, formdata];
```
Comment (shepda) [inline app/steps/ui/declaration/index.js]: Can you add a unit test for `getContext()` and `action()`.

It doesn't appear that this class has any unit tests so you'll need to create the unit test file...don't worry about unit testing the methods that you've not changed.

**Comment by bmj-vsalivendra** [comment]
Closing because preview is not created.Another PR will be crated with PREVIEW.

**Pull request closed without merging**